modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQueryParser.java

```diff
@@ -1181,8 +1181,12 @@ private GridSqlCreateTable parseCreateTable(CreateTable createTbl) {
 
         LinkedHashMap<String, GridSqlColumn> cols = new LinkedHashMap<>(data.columns.size());
 
-        for (Column col : data.columns)
+        for (Column col : data.columns) {
+            if (cols.containsKey(col.getName()))
+                throw new IgniteSQLException("Duplicate column name: " + col.getName(), IgniteQueryErrorCode.PARSING);
```

Comment:
do we have specific code for such case or may be we could add it?

Reply:
There are exists TABLE_DROP_FAILED.
So I added TABLE_CREATE_FAILED.

Reply:
Seems your first varian was better. Let's return PARSING error.

Reply:
Done